src/Oro/Bundle/CacheBundle/EventListener/CacheWarmerListener.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Oro\Bundle\CacheBundle\EventListener;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+use Oro\Component\Config\Dumper\ConfigMetadataDumperInterface;
+use Oro\Bundle\CacheBundle\Provider\ConfigCacheWarmerInterface;
+
+class CacheWarmerListener
+{
+    /** @var \SplObjectStorage */
+    protected $cacheMap;
+
+    /** @var bool */
+    protected $kernelDebug;
+
+    /**
+     * @param bool $kernelDebug
+     */
+    public function __construct($kernelDebug = false)
+    {
+        $this->kernelDebug = $kernelDebug;
+        $this->cacheMap = new \SplObjectStorage();
+    }
+
+    /**
+     * @param ConfigCacheWarmerInterface $configProvider
+     * @param ConfigMetadataDumperInterface $dumper
+     */
+    public function addConfigCacheProvider(
+        ConfigCacheWarmerInterface $configProvider,
+        ConfigMetadataDumperInterface $dumper
+    ) {
+        $this->cacheMap->attach($configProvider, $dumper);
+    }
+
+    /**
+     * Check cache metadata and warm up cache if needed
+     */
+    public function onKernelRequest()
+    {
+        if (false === $this->kernelDebug) {
```

Comment:
I think it will be better to remove this check and register this service only if debug mode is enabled.
Here is an example: https://github.com/laboro/dev/blob/master/package/platform/src/Oro/Bundle/ApiBundle/DependencyInjection/OroApiExtension.php#L74-L75